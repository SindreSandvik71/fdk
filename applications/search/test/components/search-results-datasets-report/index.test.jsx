import React from 'react';
import { shallow } from 'enzyme';
import ResultsDatasetsReport from '../../../src/components/search-results-datasets-report';
import aggregateDataset from '../../fixtures/aggregateDataset';

test('should render ResultsDatasetsReport correctly with props', () => {
  const wrapper = shallow(<ResultsDatasetsReport />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ResultsDatasetsReport correctly with hits', () => {
  const wrapper = shallow(<ResultsDatasetsReport />);
  wrapper.setState({
    aggregateDataset
  });
  expect(wrapper).toMatchSnapshot();
});
