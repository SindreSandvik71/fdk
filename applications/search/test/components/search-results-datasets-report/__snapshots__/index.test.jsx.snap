// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ResultsDatasetsReport correctly with hits 1`] = `
<div>
  <div
    className="container"
  >
    <section
      id="resultPanel"
    >
      <div
        className="row"
      >
        <div
          className="col-md-4 col-md-offset-8"
          id="content"
          role="main"
        >
          <div
            className="pull-right"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="search-filters col-sm-4 flex-move-first-item-to-bottom"
        >
          <button
            className="fdk-button fdk-button-default-no-hover"
            onClick={[Function]}
            type="button"
          >
            Fjern alle filtre
          </button>
          <SearchPublishers
            onChange={[Function]}
            onSearch={[Function]}
          />
          <SearchPublishersTree
            key="null"
            onSearch={[Function]}
            orgPath=""
          />
        </div>
        <div
          className="col-sm-8"
          id="datasets"
        >
          <ReportStats
            aggregateDataset={
              Object {
                "_shards": Object {
                  "failed": 0,
                  "successful": 5,
                  "total": 5,
                },
                "aggregations": Object {
                  "accessRightsCount": Object {
                    "buckets": Array [
                      Object {
                        "doc_count": 463,
                        "key": "PUBLIC",
                      },
                      Object {
                        "doc_count": 52,
                        "key": "RESTRICTED",
                      },
                      Object {
                        "doc_count": 32,
                        "key": "Ukjent",
                      },
                    ],
                    "doc_count_error_upper_bound": 0,
                    "sum_other_doc_count": 0,
                  },
                  "distCount": Object {
                    "doc_count": 442,
                  },
                  "distOnPublicAccessCount": Object {
                    "doc_count": 429,
                  },
                  "firstHarvested": Object {
                    "buckets": Object {
                      "last30days": Object {
                        "doc_count": 547,
                      },
                      "last365days": Object {
                        "doc_count": 547,
                      },
                      "last7days": Object {
                        "doc_count": 547,
                      },
                    },
                  },
                  "lastChanged": Object {
                    "buckets": Object {
                      "last30days": Object {
                        "doc_count": 547,
                      },
                      "last365days": Object {
                        "doc_count": 547,
                      },
                      "last7days": Object {
                        "doc_count": 547,
                      },
                    },
                  },
                  "missingFirstHarvested": Object {
                    "doc_count": 0,
                  },
                  "missingLastChanged": Object {
                    "doc_count": 0,
                  },
                  "orgPath": Object {
                    "buckets": Array [
                      Object {
                        "doc_count": 478,
                        "key": "/STAT",
                      },
                      Object {
                        "doc_count": 95,
                        "key": "/STAT/972417874",
                      },
                      Object {
                        "doc_count": 92,
                        "key": "/STAT/972417858",
                      },
                      Object {
                        "doc_count": 91,
                        "key": "/STAT/972417858/971040238",
                      },
                      Object {
                        "doc_count": 75,
                        "key": "/STAT/972417874/988983837",
                      },
                      Object {
                        "doc_count": 70,
                        "key": "/STAT/912660680",
                      },
                      Object {
                        "doc_count": 61,
                        "key": "/KOMMUNE",
                      },
                      Object {
                        "doc_count": 59,
                        "key": "/STAT/972417882",
                      },
                      Object {
                        "doc_count": 54,
                        "key": "/STAT/972417807",
                      },
                      Object {
                        "doc_count": 49,
                        "key": "/KOMMUNE/958935420",
                      },
                      Object {
                        "doc_count": 49,
                        "key": "/STAT/977161630",
                      },
                      Object {
                        "doc_count": 36,
                        "key": "/STAT/972417807/974761076",
                      },
                      Object {
                        "doc_count": 33,
                        "key": "/STAT/977161630/970205039",
                      },
                      Object {
                        "doc_count": 27,
                        "key": "/STAT/912660680/974760673",
                      },
                      Object {
                        "doc_count": 26,
                        "key": "/STAT/912660680/970188290",
                      },
                      Object {
                        "doc_count": 26,
                        "key": "/STAT/972417882/999601391",
                      },
                      Object {
                        "doc_count": 25,
                        "key": "/STAT/972417882/971022264",
                      },
                      Object {
                        "doc_count": 22,
                        "key": "/KOMMUNE/958935420/974770482",
                      },
                      Object {
                        "doc_count": 22,
                        "key": "/KOMMUNE/958935420/974770482/971040823",
                      },
                      Object {
                        "doc_count": 18,
                        "key": "/STAT/972417807/971526920",
                      },
                      Object {
                        "doc_count": 18,
                        "key": "/STAT/972417874/981544315",
                      },
                      Object {
                        "doc_count": 17,
                        "key": "/STAT/972417904",
                      },
                      Object {
                        "doc_count": 16,
                        "key": "/STAT/872417842",
                      },
                      Object {
                        "doc_count": 16,
                        "key": "/STAT/977161630/870917732",
                      },
                      Object {
                        "doc_count": 13,
                        "key": "/STAT/972417904/874783242",
                      },
                      Object {
                        "doc_count": 11,
                        "key": "/KOMMUNE/940101808",
                      },
                      Object {
                        "doc_count": 10,
                        "key": "/KOMMUNE/958935420/974770474",
                      },
                      Object {
                        "doc_count": 10,
                        "key": "/STAT/983887406",
                      },
                      Object {
                        "doc_count": 9,
                        "key": "/STAT/912660680/971203420",
                      },
                      Object {
                        "doc_count": 8,
                        "key": "/KOMMUNE/958935420/974770474/996922766",
                      },
                      Object {
                        "doc_count": 8,
                        "key": "/STAT/872417842/960885406",
                      },
                      Object {
                        "doc_count": 8,
                        "key": "/STAT/972417882/974760819",
                      },
                      Object {
                        "doc_count": 8,
                        "key": "/STAT/983887406/915933149",
                      },
                      Object {
                        "doc_count": 6,
                        "key": "/PRIVAT",
                      },
                      Object {
                        "doc_count": 6,
                        "key": "/STAT/872417842/971274042",
                      },
                      Object {
                        "doc_count": 6,
                        "key": "/STAT/912660680/971349077",
                      },
                      Object {
                        "doc_count": 4,
                        "key": "/PRIVAT/985198292",
                      },
                      Object {
                        "doc_count": 4,
                        "key": "/STAT/972417823",
                      },
                      Object {
                        "doc_count": 4,
                        "key": "/STAT/972417831",
                      },
                      Object {
                        "doc_count": 4,
                        "key": "/STAT/972417831/974760983",
                      },
                      Object {
                        "doc_count": 4,
                        "key": "/STAT/972417904/971032081",
                      },
                      Object {
                        "doc_count": 4,
                        "key": "/STAT/983887457",
                      },
                      Object {
                        "doc_count": 4,
                        "key": "/STAT/983887457/889640782",
                      },
                      Object {
                        "doc_count": 3,
                        "key": "/STAT/972417823/975950662",
                      },
                      Object {
                        "doc_count": 2,
                        "key": "/ANNET",
                      },
                      Object {
                        "doc_count": 2,
                        "key": "/ANNET/Fylkeskommunene",
                      },
                      Object {
                        "doc_count": 2,
                        "key": "/KOMMUNE/958935420/974770474/976820088",
                      },
                      Object {
                        "doc_count": 2,
                        "key": "/STAT/917082308",
                      },
                      Object {
                        "doc_count": 2,
                        "key": "/STAT/972417874/985399077",
                      },
                      Object {
                        "doc_count": 2,
                        "key": "/STAT/983887406/983544622",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "/KOMMUNE/938772924",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "/PRIVAT/985821585",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "/PRIVAT/991609407",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "/STAT/872417842/970018131",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "/STAT/872417842/974767880",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "/STAT/912660680/971526157",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "/STAT/912660680/974760282",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "/STAT/966056258",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "/STAT/972417823/986105174",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "/STAT/972417858/991825827",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "/STAT/972417866",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "/STAT/972417866/961181399",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "/STAT/983887406/983544622/867668292",
                      },
                    ],
                    "doc_count_error_upper_bound": 0,
                    "sum_other_doc_count": 0,
                  },
                  "publisherCount": Object {
                    "buckets": Array [
                      Object {
                        "doc_count": 91,
                        "key": "Kartverket",
                      },
                      Object {
                        "doc_count": 75,
                        "key": "NIBIO - NORSK INSTITUTT FOR BIOØKONOMI",
                      },
                      Object {
                        "doc_count": 36,
                        "key": "SKATTEETATEN",
                      },
                      Object {
                        "doc_count": 33,
                        "key": "NORGES VASSDRAGS- OG ENERGIDIREKTORAT (NVE)",
                      },
                      Object {
                        "doc_count": 27,
                        "key": "Brønnøysundregistrene",
                      },
                      Object {
                        "doc_count": 26,
                        "key": "MILJØDIREKTORATET",
                      },
                      Object {
                        "doc_count": 26,
                        "key": "NORGES GEOLOGISKE UNDERSØKELSE",
                      },
                      Object {
                        "doc_count": 25,
                        "key": "NORSK POLARINSTITUTT",
                      },
                      Object {
                        "doc_count": 22,
                        "key": "OSLO KOMMUNE PLAN- OG BYGNINGSETATEN",
                      },
                      Object {
                        "doc_count": 18,
                        "key": "LANDBRUKSDIREKTORATET",
                      },
                      Object {
                        "doc_count": 18,
                        "key": "SSB",
                      },
                      Object {
                        "doc_count": 17,
                        "key": "OSLO KOMMUNE",
                      },
                      Object {
                        "doc_count": 16,
                        "key": "OLJEDIREKTORATET",
                      },
                      Object {
                        "doc_count": 13,
                        "key": "KYSTVERKET",
                      },
                      Object {
                        "doc_count": 11,
                        "key": "TROMSØ KOMMUNE",
                      },
                      Object {
                        "doc_count": 9,
                        "key": "FISKERIDIREKTORATET",
                      },
                      Object {
                        "doc_count": 8,
                        "key": "DIREKTORATET FOR E-HELSE",
                      },
                      Object {
                        "doc_count": 8,
                        "key": "OSLO KOMMUNE BYMILJØETATEN",
                      },
                      Object {
                        "doc_count": 8,
                        "key": "RIKSANTIKVAREN",
                      },
                      Object {
                        "doc_count": 8,
                        "key": "STATENS LÅNEKASSE FOR UTDANNING",
                      },
                      Object {
                        "doc_count": 6,
                        "key": "HAVFORSKNINGSINSTITUTTET",
                      },
                      Object {
                        "doc_count": 6,
                        "key": "METEOROLOGISK INSTITUTT",
                      },
                      Object {
                        "doc_count": 4,
                        "key": "AVINOR AS",
                      },
                      Object {
                        "doc_count": 4,
                        "key": "DIREKTORATET FOR SAMFUNNSSIKKERHET OG BEREDSKAP (DSB)",
                      },
                      Object {
                        "doc_count": 4,
                        "key": "NAV",
                      },
                      Object {
                        "doc_count": 4,
                        "key": "STATENS VEGVESEN",
                      },
                      Object {
                        "doc_count": 3,
                        "key": "FORSVARSBYGG",
                      },
                      Object {
                        "doc_count": 2,
                        "key": "BANE NOR SF",
                      },
                      Object {
                        "doc_count": 2,
                        "key": "Fylkeskommunene",
                      },
                      Object {
                        "doc_count": 2,
                        "key": "MATTILSYNET",
                      },
                      Object {
                        "doc_count": 2,
                        "key": "OSLO KOMMUNE RENOVASJONSETATEN",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "ARKIVVERKET",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "DIFI",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "DIREKTORATET FOR MINERALFORVALTNING MED BERGMESTEREN FOR SVALBARD",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "FORSVARET",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "HELSEDIREKTORATET",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "HITRA KOMMUNE",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "NORGES TEKNISK-NATURVITENSKAPELIGE UNIVERSITET NTNU",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "PATENTSTYRET",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "RUTER AS",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "STATENS STRÅLEVERN",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "STATSKOG SF",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "UNINETT NORID AS",
                      },
                      Object {
                        "doc_count": 1,
                        "key": "UTDANNINGSDIREKTORATET",
                      },
                    ],
                    "doc_count_error_upper_bound": 0,
                    "sum_other_doc_count": 0,
                  },
                  "subjectCount": Object {
                    "doc_count": 9,
                  },
                  "theme_count": Object {
                    "buckets": Array [
                      Object {
                        "doc_count": 178,
                        "key": "ENVI",
                      },
                      Object {
                        "doc_count": 148,
                        "key": "GOVE",
                      },
                      Object {
                        "doc_count": 117,
                        "key": "REGI",
                      },
                      Object {
                        "doc_count": 114,
                        "key": "AGRI",
                      },
                      Object {
                        "doc_count": 31,
                        "key": "HEAL",
                      },
                      Object {
                        "doc_count": 27,
                        "key": "SOCI",
                      },
                      Object {
                        "doc_count": 27,
                        "key": "TRAN",
                      },
                      Object {
                        "doc_count": 24,
                        "key": "ENER",
                      },
                      Object {
                        "doc_count": 18,
                        "key": "EDUC",
                      },
                      Object {
                        "doc_count": 18,
                        "key": "JUST",
                      },
                      Object {
                        "doc_count": 8,
                        "key": "ECON",
                      },
                      Object {
                        "doc_count": 7,
                        "key": "Ukjent",
                      },
                    ],
                    "doc_count_error_upper_bound": 0,
                    "sum_other_doc_count": 0,
                  },
                },
                "hits": Object {
                  "hits": Array [],
                  "max_score": 0,
                  "total": 547,
                },
                "timed_out": false,
                "took": 10,
              }
            }
            entity=""
          />
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`should render ResultsDatasetsReport correctly with props 1`] = `
<div>
  <div
    className="container"
  >
    <section
      id="resultPanel"
    >
      <div
        className="row"
      >
        <div
          className="col-md-4 col-md-offset-8"
          id="content"
          role="main"
        >
          <div
            className="pull-right"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="search-filters col-sm-4 flex-move-first-item-to-bottom"
        >
          <button
            className="fdk-button fdk-button-default-no-hover"
            onClick={[Function]}
            type="button"
          >
            Fjern alle filtre
          </button>
          <SearchPublishers
            onChange={[Function]}
            onSearch={[Function]}
          />
          <SearchPublishersTree
            key="null"
            onSearch={[Function]}
            orgPath=""
          />
        </div>
        <div
          className="col-sm-8"
          id="datasets"
        >
          <ReportStats
            aggregateDataset={Object {}}
            entity=""
          />
        </div>
      </div>
    </section>
  </div>
</div>
`;
